src/lib/configs/project/dockerCompose.ts

```diff
@@ -329,3 +403,96 @@ export const initReadonlyDockerComposeConfig =
   getReadonlyConfigInitFunction(initConfigOptions);
 
 export const dockerComposeSchema: JSONSchemaType<LatestConfig> = configSchemaV0;
+
+export async function ensureConfigToml(
+  fluenceConfig: FluenceConfigReadonly | null,
+  providerConfig: ProviderConfigReadonly,
+) {
+  const baseNoxConfig = mergeNoxConfigYAML(
+    await getDefaultNoxConfigYAML(fluenceConfig),
+    providerConfig.nox ?? {},
+  );
+
+  const configsDir = await ensureFluenceConfigsDir();
+  const { stringify } = await import("@iarna/toml");
+
+  await Promise.all(
+    Object.entries(providerConfig.computePeers).map(async ([key, value], i) => {
+      const overridden = mergeNoxConfigYAML(baseNoxConfig, value.nox ?? {});
+
+      if (overridden.tcpPort === undefined) {
+        overridden.tcpPort = TCP_PORT_START + i;
+      }
+
+      if (overridden.websocketPort === undefined) {
+        overridden.websocketPort = WEB_SOCKET_PORT_START + i;
+      }
+
+      if (overridden.httpPort === undefined) {
+        overridden.httpPort = HTTP_PORT_START + i;
+      }
+
+      if (
+        "systemServices" in overridden &&
+        "decider" in overridden.systemServices &&
+        overridden.systemServices.decider.walletKey === undefined
```

Comment:
I don't understand the logic here.

Why not like this?
```
if overridden?.systemServices?.decider?.walletKey === undefined
```

In this case, it would write wallet key even if there are no `systemServices` or `systemServices.decider` in the `overridden`. But right now it doesn't, and I'm not sure why.

Reply:
I'v been thinking that there is probably a chance that user will not want to have decider at all for the local test.
If we do it like you are suggesting - then this property will be set in all cases. I don't know, Maybe we can do it like you suggest, cause we probably better of not thinking about the cases when there is no decider. What do you think?

Reply:
Yeah, I think that Decider should be enabled by default. Otherwise `deal deploy` won't work.